crates/swc_ecma_minifier/src/compress/optimize/sequences.rs

```diff
@@ -1597,6 +1597,33 @@ where
                     return Ok(false);
                 }
 
+                // We can't merge into `[]` in some cases because `obj` is **resolved** before
+                // evaluating `[]`.
+                //
+                // See https://github.com/swc-project/swc/pull/6509
+
+                // We check only for an identifier because resolution is the problematic part.
+                // Evalutation order is not a problem.
```

Comment:
Is this true? Wouldn't `(1, obj)[obj = {}, 'key']` hit the same problem?

Reply:
Oh, I was wrong. Nice catch, and thank you!

Reply:
Nevermind, 
```js
var obj;
console.log((1, obj)[obj = {}, 'key'])
```

is expected to fail even without minification.